JRC Publications Repository.js

```diff
@@ -0,0 +1,477 @@
+		//e.g. <meta name="DC.creator" content="DEL RIO GONZALEZ Pablo" xml:lang="en_GB">
+		//We try to overwrite the creators with better splitting of lastName and firstName.
+		var authors = ZU.xpath(doc, '//meta[@name="DC.creator" and @content]|//meta[@name="DC.contributor" and @content]');
+		if (authors) {
+			item.creators = [];
+			for (var a=0; a<authors.length; a++) {
+				var authorsText = authors[a].content;
+				var authorsType = authors[a].name.substr(3);//either creator or contributer
+				var lastName = "";
+				var firstName = "";
+				var authorParts = authorsText.split(' ');
+				for (var b=0; b<authorParts.length; b++) {
+					//distinguish between lastName (every letter is in uppercase) from firstName
+					//but there might also be just initials (e.g. "D.") from the firstName
+					if (authorParts[b].toUpperCase() == authorParts[b] && authorParts[b].indexOf('.') == -1) {
+						lastName += ' ' + ZU.capitalizeTitle(authorParts[b], true)
+					} else {
+						firstName += ' ' + authorParts[b];
+					}
+				}
+				if (authorsType == "creator") {
+					item.creators.push( {lastName:lastName.trim(), firstName:firstName.trim(), creatorType:"author" });
+				} else {
+					item.creators.push( {lastName:lastName.trim(), firstName:firstName.trim(), creatorType:"contributor" });
+				}
+				
+			}
+		}
+
+		//volume, issue, pages
+		var citation = ZU.xpathText(doc, '//meta[contains(@name, "DCTERMS.bibliographicCitation")]/@content');
+		//e.g. GLOBAL ENVIRONMENTAL CHANGE-HUMAN AND POLICY DIMENSIONS vol. 23 no. 5 p. 892-901
+		//e.g. EUROPEAN PHYSICAL JOURNAL C vol. 73 no. 3 p. 2330 [29 pages]
+		if (citation) {
+			var seperationPos = citation.lastIndexOf('vol.');
+			if (seperationPos == -1) {
+				seperationPos = citation.lastIndexOf('no.');
+			}
+			if (seperationPos == -1) {
+				seperationPos = citation.lastIndexOf('p.');
+			}
+			
+			if (seperationPos > -1) {
+				item.publicationTitle = citation.substring(0,seperationPos).trim();
+				citation = citation.substring(seperationPos);
+				var volume = citation.match(/vol\. (\d+)/);
+				if (!item.volume && volume && volume.length) {
+					item.volume = volume[1];
+				}
+				var issue = citation.match(/no\. (\d+)/);
+				if (!item.issue && issue && issue.length) {
+					item.issue = issue[1];
+				}
+				var pages = citation.match(/p\. (\d+\s*-?\s*\d*)/);
+				if (!item.pages && pages && pages.length) {
+					item.pages = pages[1];
+				}
+			} else {
+				item.publicationTitle = citation;
+			}
+			if (item.publicationTitle == item.publicationTitle.toUpperCase() ) {
+				item.publicationTitle = ZU.capitalizeTitle(item.publicationTitle, true );
+			}
+			
+		}
+		
+		//The itemType is saved as a tag, which is not useful. Thus we delete it.
+		//e.g. <meta name="citation_keywords" content="PhD Theses">
+		item.tags = [];
+		//e.g. <meta name="DC.description" content="JRC.H.5-Land Resources Management" xml:lang="en_GB">
+		item.tags.push(item.extra);
```

Comment:
I wouldn't rely on `item.extra` being anything specific. This can change at any time. Fetch directly from page